File: tasks/day-1/erlang/main.erl
-module(main).
-export([start/1]).

start([InputFile]) ->
    {ok, Data} = file:read_file(InputFile),
    {Level, Basemant, _} = lists:foldl(fun level/2, {0, 0, 1}, binary_to_list(Data)),
    io:format("Level: ~w~nBasemant: ~w~n", [Level, Basemant]),
    halt().

level(Char, {-1, 0, Index}) ->
    level(Char, {-1, Index - 1, Index});
level($(, {Level, Basemant, Index}) -> 
    {Level + 1, Basemant, Index + 1};
level($), {Level, Basemant, Index}) -> 
    {Level - 1, Basemant, Index + 1};
level(_, Acc) -> Acc.

File: tasks/day-1/python/main.py
import sys

file_path = sys.argv[1]

file = open(file_path)
content = file.read()

level, basemant = 0, 0

for index, char in enumerate(content):
	if char not in ('(', ')'):
		continue

	level = level + 1 if char == '(' else level - 1
	if not basemant and level < 0:
		basemant = index + 1

print('Level: {}\nBasemant: {}'.format(level, basemant))